tests/database.test.js

```diff
@@ -0,0 +1,124 @@
+const app = require('../server/app.js');
+const supertest = require("supertest");
+const http = require('http');
+const mongoose = require('mongoose');
+const User = require('../server/dbSchemas/attendanceSchema.js').User;
+const Course = require('../server/dbSchemas/attendanceSchema.js').Course;
+const SeatingLayout = require('../server/dbSchemas/attendanceSchema.js').SeatingLayout;
+
+describe('Database Functionality', () => {
+
+    let request;
+    let server;
+
+    // Open database before running tests
+    beforeAll(async (done) => {
+        server = http.createServer(app);
+        server.listen();
+        request = supertest(server);
+        const db = mongoose.connection;
+        db.once('open', done);
+    });
+
+    // Close database & server when done
+    afterAll(async (done) => {
+        await mongoose.connection.close();
+        await server.close(done);
+    });
+
+    it('Should store user to database', async done => {
+        //Send request to register user
+        const res = await request.post('/api/register').send({
+            email: 'dummy@test.com',
+            name: 'dummy test',
+            password:'12345',
+            is_professor: true
+        });
+
+        //Search user in database by email
+        const user = await User.findOne({
+            email: 'dummy@test.com'
+        });
+
+        //Check that user has name, email, password, and is or is not a professor
+        expect(user.name).toBeTruthy()
+        expect(user.email).toBeTruthy()
+        expect(user.password).toBeTruthy()
+        expect(typeof(user.is_professor) === 'boolean').toBeTruthy();
+
+        //Delete test user after testing
+        response = await request.delete("/api/delete-user").send({
+            email: 'dummy@test.com'
+        });
+
+        done ()
+    });
+
+    xit('Should store course to database', async done => {
+
+        //Create test course
+        const res = await request.post('/api/createSection').send({
+            name: 'SWE4103',
+            admin: 'Dr. MacIsaac',
+            professor: 'Dawn MacIsaac',
+            maxCapacity: 30,
+            attendance: [Date, [], []],
+            date_created: Date
```

Comment:
The db schema changed so the attendance has a `mandatory` true/false field and doesn't have a `date_created` field. The schema has been finalized and merged so you can see it in the `create-section-story` branch [here](https://github.com/fredy20501/AttendanceTracker/blob/create-section-story/server/dbSchemas/attendanceSchema.js).

Reply:
Okay, I'll change that.